src/script/chem/struct/index.js

```diff
@@ -44,6 +44,11 @@ function Struct() {
 	this.sGroupForest = new SGroupForest(this);
 }
 
+/** @return {?Atom} */
+Struct.prototype.hasExplicitHydrogen = function () {
```

Comment:
this method is going to return the first atom, that is a Hydrogen - which doesn't seem consistent with the method name. Maybe you want to return a boolean, which tells whether the molecule comprises of at least one hydrogen atom?

Reply:
agreed thanks